src/audio/SDL_audiocvt.c

```diff
@@ -157,15 +157,96 @@ static SDL_bool SDL_IsSupportedChannelCount(const int channels)
     return ((channels >= 1) && (channels <= 8));
 }
 
+SDL_bool SDL_ChannelMapIsBogus(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = 0; i < channels; i++) {
+            if (map[i] >= ((Uint8) channels)) {
+                return SDL_TRUE;
+            }
+        }
+    }
+    return SDL_FALSE;
+}
+
+SDL_bool SDL_ChannelMapIsDefault(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = 0; i < channels; i++) {
+            if (map[i] != i) {
+                return SDL_FALSE;
+            }
+        }
+    }
+    return SDL_TRUE;
+}
+
+SDL_bool SDL_ChannelMapHasDuplicates(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = channels - 1; i >= 1; i--) {
+            for (int j = i - 1; j >= 0; j--) {
+                if (map[i] == map[j]) {
+                    return SDL_TRUE;
+                }
+            }
+        }
```

Comment:
Small performance nit, this could use a bitset to keep track of the seen channels rather than scanning the map.
```c
Uint32 seen = 0;

for (int i = 0; i < channels; ++i) {
    Uint32 mask = (Uint32)1 << map[i];

    if (seen & mask) {
        return SDL_TRUE;
    }

    seen |= mask;
}
```

Reply:
Good idea, but this function is gone in 136a33153cc797eeed2b46600638c76bb064e6a1, so we're good either way.  :)